File: 1872-can-you-eat-your-favorite-candy-on-your-favorite-day/1872-can-you-eat-your-favorite-candy-on-your-favorite-day.py
class Solution:
    def canEat(self, candiesCount: List[int], queries: List[List[int]]) -> List[bool]:
        n = len(candiesCount)
        res = [0] * len(queries)
        pref =  [0] * (n + 1) 

        for i in range(1, n + 1): 
            pref[i] = pref[i - 1] + candiesCount[i - 1]

        for i in range(len(queries)): 
            t = queries[i][0]
            time = queries[i][1]
            limit = queries[i][2]

            maxi = pref[t + 1] - 1
            mini = pref[t] // limit

            if (time <= maxi and time >= mini): 
                res[i] = True 
            else: 
                res[i] = False 

        return res
            

File: 1872-can-you-eat-your-favorite-candy-on-your-favorite-day/1872-can-you-eat-your-favorite-candy-on-your-favorite-day.java
class Solution {
    private long pref[]; 

    public boolean[] canEat(int[] candiesCount, int[][] queries) {
        int n = candiesCount.length;
        pref = new long[n + 1];
        
        for (int i = 1; i <= n; i++) 
            pref[i] = pref[i - 1] + candiesCount[i - 1] * 1L;

        boolean res[] = new boolean[queries.length];
        for (int i = 0; i < queries.length; i++) {
            int type = queries[i][0], time = queries[i][1], limit = queries[i][2];
            long maxi = pref[type + 1] - 1, mini = pref[type] / limit;
            if (time <= maxi && time >= mini) 
                res[i] = true;
            else 
                res[i] = false;
        }  
        
        return res; 
    }
}